These defines are no longer available with ncurses-5.7.

--- origsrc/libcaca-0.99.beta16/caca/driver/ncurses.c	2008-09-27 10:58:05.000000000 -0500
+++ src/libcaca-0.99.beta16/caca/driver/ncurses.c	2009-03-29 17:24:09.531250000 -0500
@@ -463,28 +463,24 @@ static int ncurses_get_event(caca_displa
             case BUTTON1_CLICKED: CLICK(1); break;
             case BUTTON1_DOUBLE_CLICKED: CLICK(1); CLICK(1); break;
             case BUTTON1_TRIPLE_CLICKED: CLICK(1); CLICK(1); CLICK(1); break;
-            case BUTTON1_RESERVED_EVENT: break;
 
             case BUTTON2_PRESSED: PRESS(2); break;
             case BUTTON2_RELEASED: RELEASE(2); break;
             case BUTTON2_CLICKED: CLICK(2); break;
             case BUTTON2_DOUBLE_CLICKED: CLICK(2); CLICK(2); break;
             case BUTTON2_TRIPLE_CLICKED: CLICK(2); CLICK(2); CLICK(2); break;
-            case BUTTON2_RESERVED_EVENT: break;
 
             case BUTTON3_PRESSED: PRESS(3); break;
             case BUTTON3_RELEASED: RELEASE(3); break;
             case BUTTON3_CLICKED: CLICK(3); break;
             case BUTTON3_DOUBLE_CLICKED: CLICK(3); CLICK(3); break;
             case BUTTON3_TRIPLE_CLICKED: CLICK(3); CLICK(3); CLICK(3); break;
-            case BUTTON3_RESERVED_EVENT: break;
 
             case BUTTON4_PRESSED: PRESS(4); break;
             case BUTTON4_RELEASED: RELEASE(4); break;
             case BUTTON4_CLICKED: CLICK(4); break;
             case BUTTON4_DOUBLE_CLICKED: CLICK(4); CLICK(4); break;
             case BUTTON4_TRIPLE_CLICKED: CLICK(4); CLICK(4); CLICK(4); break;
-            case BUTTON4_RESERVED_EVENT: break;
 
             default:
                 break;
