--- origsrc/libcaca-0.99.beta16/ruby/Makefile.am	2008-04-25 11:36:15.000000000 -0500
+++ src/libcaca-0.99.beta16/ruby/Makefile.am	2008-08-06 12:23:40.656250000 -0500
@@ -25,7 +25,7 @@
                   caca-event.c \
                   caca-event.h \
                   $(NULL)
-caca_la_LDFLAGS = -module -avoid-version -shared $(RUBY_LIBS)
+caca_la_LDFLAGS = -module -avoid-version -no-undefined $(RUBY_LIBS) -shrext .so
 caca_la_LIBADD = ../caca/libcaca.la
 
 EXTRA_DIST = ruby.dox \
